src/main/java/appeng/container/implementations/ContainerCraftConfirm.java

```diff
@@ -175,6 +175,12 @@ public void detectAndSendChanges() {
                             missing.setStackSize(missing.getStackSize() - toExtract.getStackSize());
                         }
 
+                        if (toExtract.getStackSize() > 0 && toCraft.getStackSize() <= 0 && missing == null
+                                || (missing != null && missing.getStackSize() <= 0)) {
+                            long available = items.getAvailableItem(toExtract).getStackSize();
+                            toExtract.setUsedPercent(toExtract.getStackSize() / (available / 100f));
```

Comment:
is available always nonzero value?

Reply:
Yes, because if there are no stored items it shows up as missing and `long available = items.getAvailableItem(toExtract).getStackSize();` is not reached